aiocogeo_tiler/tiler.py

```diff
@@ -48,27 +49,16 @@ class COGInfo:
     colormap: Optional[Dict[int, Sequence[int]]] = None
 
 
-@dataclass
-class TileResponse:
+class TileResponse(ImageData):
     """tile response"""
 
-    arr: np.ndarray
-    mask: Optional[np.ma.masked_array] = None
-
-    def __iter__(self):
-        """
-        Allow for variable expansion (``arr, mask = TileResponse``)
-
-        """
-        for i in (self.arr, self.mask * 255):
-            yield i
-
     @classmethod
     def from_array(cls, array: Union[np.ndarray, np.ma.masked_array]) -> "TileResponse":
         """create from numpy array"""
         if isinstance(array, np.ma.masked_array):
-            return cls(array.data, array.mask)
-        return cls(array)
+            return cls(data=array.data, mask=(array.mask * 255).astype("uint8"))
+        mask = np.zeros((array.shape[0], array.shape[1]), dtype="uint8") + 255
```

Comment:
with rio-tiler==2.0.0rc2 I thin this can go

Reply:
yeah :)